industrial_robot_client/src/joint_trajectory_action.cpp

```diff
@@ -224,7 +226,7 @@ void JointTrajectoryAction::controllerStateCB(const control_msgs::FollowJointTra
     return;
   }
 
-  if (ros::Time::now() < time_to_check_)
+  if (!has_moved_once_ && (ros::Time::now() < time_to_check_))
```

Comment:
well, even if has_moved_once_ is never set, the motion will succeed at either time_to_check if the robot is already at the goal or at same point later when it arrives there. So it is still compatible with those old drivers.

How would you do it in the old way? I don't know of any other information that could be used then.